Source/WebCore/page/DOMWindow.cpp

```diff
@@ -922,4 +922,124 @@ ExceptionOr<PushManager&> DOMWindow::pushManager()
 }
 #endif
 
+bool DOMWindow::isCurrentlyDisplayedInFrame() const
+{
+    RefPtr frame = this->frame();
+    return frame && frame->window() == this;
+}
+
+void DOMWindow::printErrorMessage(const String& message) const
+{
+    if (message.isEmpty())
+        return;
+
+    if (CheckedPtr pageConsole = console())
+        pageConsole->addMessage(MessageSource::JS, MessageLevel::Error, message);
+}
+
+String DOMWindow::crossDomainAccessErrorMessage(const LocalDOMWindow& activeWindow, IncludeTargetOrigin includeTargetOrigin)
+{
+    const URL& activeWindowURL = activeWindow.document()->url();
+    if (activeWindowURL.isNull())
+        return String();
+
+    RefPtr<Document> document;
+    if (this->document().hasException())
```

Comment:
This always returns an exception if the DOMWindow is a RemoteDOMWindow.  I'm not sure that's what we want.